new(gatsby-plugin-vanilla-extract): add to repo (#148)
gatsby-uc/plugins: packages/gatsby-plugin-vanilla-extract/README.md

# gatsby-plugin-vanilla-extract

Gatsby plugin which wraps the [vanilla-extract](https://vanilla-extract.style/) project for easy use in Gatsby.

## Setup

> `npm install gatsby-plugin-vanilla-extract @vanilla-extract/babel-plugin @vanilla-extract/css @vanilla-extract/webpack-plugin`

Reviewer: ````suggestion
```shell
npm install gatsby-plugin-vanilla-extract @vanilla-extract/babel-plugin @vanilla-extract/css @vanilla-extract/webpack-plugin
```
````

Always better to put install messages in big code blocks

Add to your site's `gatsby-config.js`.

```js
module.exports = {
  plugins: [`gatsby-plugin-vanilla-extract`],
};
```

> 

Reviewer: ````suggestion

You can also pass any [vanilla-extract configuration object](https://vanilla-extract.style/documentation/setup/#configuration) to the plugin.

```js
module.exports = {
  plugins: [
    {
      resolve: `gatsby-plugin-vanilla-extract`,
      options: {
        identifiers: `short`
      }
    }
  ],
};
```

````

## How to use

> See the [vanilla-extract README](https://github.com/seek-oss/vanilla-extract#readme) for full documentation.

Reviewer: ```suggestion
See the [vanilla-extract website](https://vanilla-extract.style/) for full documentation.
```

> Simple example.
> 
> `src/pages/index.tsx`

Reviewer: ```suggestion
`src/pages/index.tsx`:
```

```tsx
import * as React from "react";

import { className } from "../styles/index.css.ts";

export default function Index() {
  return <div className={className}>Pizza</div>;
}
```

> `src/styles/index.css.ts`

Reviewer: ```suggestion
`src/styles/index.css.ts`:
```

```ts
import { style } from "@vanilla-extract/css";

export const className = style({
  background: "red",
});
```

> ## Using a Starter
> Kick off your project using our [Typescript starter](https://github.com/gatsbyjs/gatsby-starter-typescript
> )

Reviewer: ```suggestion
### Example Project

Kick off your project using Gatsby's [example project](TODO)
```

That's the example you'll be creating in the monorepo
